packages/web-react/src/components/Collapse/useCollapseStyle.ts

```diff
@@ -0,0 +1,76 @@
+import { CSSProperties, MutableRefObject, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useClassNamePrefix } from '../../hooks';
+import { ClickEvent, CollapseResponsiveType, CollapseContentProps } from '../../types';
+
+export interface useCollapseStyleProps {
+  id?: string;
+  collapsed?: boolean;
+  responsive?: CollapseResponsiveType;
+  contentHeight?: string;
+  onToggle?: (event: ClickEvent) => void;
+  contentRef?: MutableRefObject<HTMLButtonElement | undefined>;
+  wrapperClassName?: string;
+  wrapperStyle?: CSSProperties;
+  contentProps?: CollapseContentProps;
+}
+
+export const useCollapseStyle = (props: useCollapseStyleProps) => {
```

Comment:
Please, stick with the implementation for an example of `useAlertStyleProps`. Just return an object with classes. You are mixing aria attributes, classes and effects in one place.